tests/test_main.py

```diff
@@ -23,69 +20,27 @@ async def app(scope, receive, send):
         pytest.param("::1", "http://[::1]:8000", id="ipv6"),
     ],
 )
-def test_run(host, url):
+async def test_run(host, url):
     config = Config(app=app, host=host, loop="asyncio", limit_max_requests=1)
-    server = Server(config=config)
-    thread = threading.Thread(target=server.run)
-    thread.start()
-    while not server.started:
-        time.sleep(0.01)
-    response = requests.get(url)
+    async with run_server(config):
+        async with httpx.AsyncClient() as client:
+            response = await client.get(url)
     assert response.status_code == 204
-    thread.join()
 
 
-def test_run_multiprocess():
+@pytest.mark.asyncio
+async def test_run_multiprocess():
     config = Config(app=app, loop="asyncio", workers=2, limit_max_requests=1)
-    server = Server(config=config)
-    thread = threading.Thread(target=server.run)
-    thread.start()
-    while not server.started:
-        time.sleep(0.01)
-    response = requests.get("http://127.0.0.1:8000")
+    async with run_server(config):
+        async with httpx.AsyncClient() as client:
+            response = await client.get("http://127.0.0.1:8000")
     assert response.status_code == 204
-    thread.join()
 
 
-def test_run_reload():
+@pytest.mark.asyncio
+async def test_run_reload():
     config = Config(app=app, loop="asyncio", reload=True, limit_max_requests=1)
-    server = Server(config=config)
-    thread = threading.Thread(target=server.run)
-    thread.start()
-    while not server.started:
-        time.sleep(0.01)
-    response = requests.get("http://127.0.0.1:8000")
+    async with run_server(config):
+        async with httpx.AsyncClient() as client:
+            response = await client.get("http://127.0.0.1:8000")
     assert response.status_code == 204
-    thread.join()
-
-
-def test_run_with_shutdown():
```

Comment:
Could you clarify why we can do without this test now?

Reply:
ok maybe it's me misunderstanding what the test does but..

the way i get it is that it's equivalent to the new `async with run-server` with the slight exception that a socket `sock` was explicitly binded, and it just checks the fact no exception is raised in that case.

Now if the context manager was raising an exception all our tests would fail. hence why I didnt bother.

honestly at first I tried to run it  "the new way" and couldnt find a way that was really testing something.

Reply:
Yes, OK, sounds good :)